cfme/tests/cloud/test_providers.py

```diff
@@ -186,6 +189,27 @@ def vm_ip(cfme_vhd, pwsh_ssh):
         pwsh_ssh.run_command("pwsh {}".format(path_script), timeout=180)
 
 
+@pytest.fixture
+def ec2_provider_with_sts_creds(appliance):
```

Comment:
I think this fixture can be simplified by first instantiating the ec2west provider and then just overriding it's endpoint attributes. 

```python,
provider = providers.get_crud("ec2west")
assume_role_creds = provider.data.sts_assume_role.credentials
creds = Credential(principal=credentials[assume_role_creds]['username'],
                       secret=credentials[assume_role_creds]['password'])
endpoint = EC2Endpoint(
        assume_role_arn=provider.data.sts_assume_role.role_arn,
        credentials=creds)
provider.endpoints = endpoint
```

What do you think?

Reply:
Reworked